planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp

```diff
@@ -188,17 +239,11 @@ bool NormalLaneChange::is_near_regulatory_element() const
 
   if (current_lanes.empty()) return false;
 
-  const auto shift_intervals =
-    getRouteHandler()->getLateralIntervalsToPreferredLane(current_lanes.back());
-
-  if (shift_intervals.empty()) return false;
-
-  const auto & lc_params = *common_data_ptr_->lc_param_ptr;
   const auto max_prepare_length = calculation::calc_maximum_prepare_length(common_data_ptr_);
-  const auto min_lc_length =
-    calculation::calc_minimum_lane_change_length(lc_params, shift_intervals);
   const auto dist_to_terminal_start =
-    calculation::calc_ego_dist_to_terminal_end(common_data_ptr_) - min_lc_length;
+    calculation::calc_dist_from_pose_to_terminal_end(
+      common_data_ptr_, current_lanes, common_data_ptr_->get_ego_pose()) -
+    common_data_ptr_->transient_data.current_dist_buffer.min;
```

Comment:
why not directly use `common_data_ptr_->transient_data.dist_from_ego_to_current_terminal_start`

Reply:
changed in https://github.com/autowarefoundation/autoware.universe/pull/8849/commits/56c45d8960f7574105c07973019378729f833fab